File: api/admin/link.go
package admin

import (
	"short-link/controller"

	"github.com/gin-gonic/gin"
)

func NewLinkRouter(rg *gin.RouterGroup) {
	linkController := controller.NewLinkController()
	blackListController := controller.NewBlackListController()

	linkGroup := rg.Group("/link")
	linkGroup.POST("/add", linkController.AddLink)
	linkGroup.POST("/list", linkController.LinkList)
	linkGroup.POST("/del", linkController.DeleteLink)

	linkBlackListGroup := linkGroup.Group("/black-list")
	linkBlackListGroup.POST("/add", blackListController.AddBlackList)
	linkBlackListGroup.POST("/del", blackListController.DeleteBlackList)
	linkBlackListGroup.POST("/list", blackListController.ListBlackList)
	linkBlackListGroup.POST("/update-status", blackListController.UpdateBlackListStatus)

}

File: front/short-link/src/services/short-link/link.ts
import {request} from "@umijs/max";


export async function addLink(data) {
    return request("http://127.0.0.1:8088/short-link/api/v1/admin/link/add", {
        method: 'POST',
        data
    })
}


export async function linkList(data) {
    return request("http://127.0.0.1:8088/short-link/api/v1/admin/link/list", {
        method: 'POST',
        data
    })
}

export async function delLink(data){
    return request("http://127.0.0.1:8088/short-link/api/v1/admin/link/del", {
        method: 'POST',
        data
    })
}
